pkg/runTest.py

```diff
@@ -0,0 +1,40 @@
+"""
+run test
+"""
+
+import logging
+from fmatch.matcher import Matcher
+from pkg.logrus import SingletonLogger
+from pkg.utils import run_hunter_analyze, load_config, get_es_url, process_test
+
+
+def run(**kwargs):
+    """run method to start the tests
+
+    Args:
+        config (_type_): file path to config file
+        output_path (_type_): output path to save the data
+        hunter_analyze (_type_): changepoint detection through hunter. defaults to True
+        output_format (_type_): output to be table or json
+
+    Returns:
+        _type_: _description_
+    """
+    logger_instance = SingletonLogger(debug=logging.INFO).logger
```

Comment:
I'm wondering if this is the issue causing: https://github.com/cloud-bulldozer/py-commons/pull/22 because it is creating a new logger each time a new post is ran in the daemon mode

I would think you would want to try to pass the logger from upper daemon mode setting to here instead of creating a new one every run and in sub functions. IE the daemon mode cli would create 1 single logger and every time orion is called it would use the same one. This would also help with [this issue](https://github.com/cloud-bulldozer/py-commons/pull/22). I might be thinking about it wrong but any thoughts? @jtaleric @shashank-boyapally @vishnuchalla

Reply:
Hi @paigerube14 , since we implemented SingletonLogger, it only creates the logger for once, later everywhere we call `SingletonLogger(debug=logging.INFO).logger` it returns back the already created logger instead of creating a new one. Also, since the context(i.e. file and line info) the logger prints is different in fmatch and orion, we could see that only logger of fmatch is duplicating and not orion. I think passing the logger to each function was redundant which was the case in our previous implementation. In order to tackle this `SingletonLogger` class was implemented.

Reply:
One more point to consider here: https://github.com/cloud-bulldozer/orion/pull/26#discussion_r1537446809